webknossos/webknossos/dataset/_utils/segmentation_recognition.py

```diff
@@ -0,0 +1,73 @@
+import random
+from pathlib import Path
+
+import numpy as np
+
+from webknossos.dataset.layer_categories import LayerCategoryType
+from webknossos.dataset.mag_view import MagView
+from webknossos.geometry.bounding_box import BoundingBox
+from webknossos.geometry.vec3_int import Vec3Int
+
+NUM_SAMPLES = 20
+THRESHOLD = (
+    1 / 200
+)  # more unique values per voxel than this value means color, less segmentation
+SAMPLE_SIZE = Vec3Int(16, 16, 16)
+MAX_FAILS = 200
+
+
+def guess_if_segmentation_path(filepath: Path) -> bool:
+    lowercase_filepath = str(filepath).lower()
+    return any(i in lowercase_filepath for i in ["segmentation", "labels"])
+
+
+def guess_category_from_view(view: MagView) -> LayerCategoryType:
+    if view.layer.num_channels > 1 or sample_distinct_values_per_vx(view) > THRESHOLD:
+        return "color"
+    return "segmentation"
+
+
+def sample_distinct_values_per_vx(view: MagView) -> float:
+    sample_size_for_view = view.bounding_box.size_xyz.pairmin(SAMPLE_SIZE * view.mag)
+    min_offset = view.bounding_box.topleft_xyz
+    max_offset = view.bounding_box.bottomright_xyz - sample_size_for_view
```

Comment:
I currently don't see, where this would fail. I think as the `sample_size_for_view` is adapted, if the view is smaller than the `SAMPLE_SIZE` constant, it should work. The main problem, when the view is too small, e.g. 12x12x12, we perform this heuristic always on the same sample. Can you explain what might be the issue?

Reply:
oh, you are right! I missed the `pairmin` bit back then. nevermind!